src/align/anchors.jl

```diff
@@ -0,0 +1,271 @@
+
+# Anchor definition
+# ------------------
+"""
+A type to store the operation enocded in an alignment (CIGAR Operations).
+Also stores the position in the alignment view of the sequences, and the
+corresponding position in the unaltered source sequence (nucleotide or protein).
+
+It stores the position values as Int types and the alignment operation is stored
+as a type `Operation`, these are defined in the file `operations.jl` and loaded
+into Bio.Align namespace as a series of global constants.
+
+Calling the AlignmentAnchor default constructor with default arguments sets both
+integer fields to 0, and the operation field to the global constant OP_INVALID.
+"""
+immutable AlignmentAnchor
+    seqpos::Int
+    refpos::Int
+    op::Operation
+end
+
+
+# Basic operators for AlignmentAnchors
+# -------------------------------------
+
+"""
+Print to screen or other IO stream, a formatted description of an
+AlignmentAnchor.
+"""
+function show(io::IO, anc::AlignmentAnchor)
```

Comment:
Now that I think about it, adding docstrings to `show` methods might not clarify anything, since the base docstring pretty much tells you what you need to know.
```
help?> show
search: show showall showerror showcompact @show @Show Cshort Cushort print_shortest searchsorted searchsortedlast searchsortedfirst flush_cstdio SharedVector

  show(x)

  Write an informative text representation of a value to the current output stream. New types should overload show(io, x) where the first argument is a stream. The representation used by show generally includes Julia-specific formatting
  and type information.

```
I don't think that's true of all Base functions. Sometimes we need more specific documentation.